tests/test_views.py

```diff
@@ -0,0 +1,97 @@
+import subjects
+from tests import factories
+from django.urls import reverse
+import pytest
+import os
+
+
+@pytest.mark.django_db
+class TestViewallSubjects:
+    def test_viewall_subjects(self, client):
+        sub_1 = factories.SubjectFactory.create(name='sub-1', parent=0, lft=1, rght=4)
+        factories.SubjectFactory.create(name='sub-2', parent=0, lft=5, rght=6)
+        factories.SubjectFactory.create(name='sub-1.1', parent=sub_1.id, lft=2, rght=3)
+        response = client.get(reverse('list'))
+        assert response.status_code == 200
+        assert response.templates[0].name == 'subjects/subject_list.html'
+        assert 'sub-1' in response.context['viewall_list'][0]['hyphenated']
+        assert 'sub-1 - sub-1.1' in response.context['viewall_list'][1]['hyphenated']
+        assert 'sub-2' in response.context['viewall_list'][2]['hyphenated']
+
+
+@pytest.mark.django_db
+class TestBrowseSubjects:
+    def test_browse_subjects(self, client):
+        s1 = factories.SubjectFactory.create()
+        s2 = factories.SubjectFactory.create()
+        response = client.get(reverse('main_browse'))
+        assert response.status_code == 200
+        assert response.templates[0].name == 'subjects/subject_browse.html'
+        assert "<a href=\'/subjects/browse/{id}/\'>{name}</a>"\
+               .format(id=s1.id, name=s1.name).encode() in response.content
+        assert "<a href=\'/subjects/browse/{id}/\'>{name}</a>"\
+               .format(id=s2.id, name=s2.name).encode() in response.content
+
+    def test_browse_subjects_with_sub_id(self, client):
+        sub = factories.SubjectFactory.create()
+        response = client.get(reverse('browse', kwargs={'sub_id': sub.id}))
+        assert response.status_code == 200
+        assert response.templates[0].name == 'subjects/subject_browse.html'
+        assert "<input name=\'subject\' type=\'text\' size=\'57\' value=\'{name}\'>".format(
+               name=sub.name).encode() in response.content
+
+    def test_browse_subjects_with_sub_name(self, client):
+        sub = factories.SubjectFactory.create()
+        response = client.get(reverse('browse_sub_name', kwargs={'sub_name': sub.name}))
+        assert response.status_code == 200
+        assert response.templates[0].name == 'subjects/subject_browse.html'
+        assert "<input name=\'subject\' type=\'text\' size=\'57\' value=\'{name}\'>".format(
+               name=sub.name).encode() in response.content
+
```

Comment:
I think Gio's comment about checking context variables instead of html make sense for the above browse tests and the following search tests.

Reply:
I asserted context variables in the other tests too.